pkg/runtime/bulk_subscriber.go

```diff
@@ -295,16 +296,19 @@ func (a *DaprRuntime) publishBulkMessageHTTP(ctx context.Context, msg *pubsubBul
 	req.WithRawData(msg.data, contenttype.CloudEventContentType)
 	req.WithCustomHTTPMetadata(msg.metadata)
 
-	for i, cloudEvent := range msg.cloudEvents {
+	n := 0
```

Comment:
This is needed because there's at least one path below (`cloudEvent[pubsub.TraceIDField] == nil`) when otherwise you skip adding an item to `spans`, causing the slice to have zero values inside. So using a separate counter, then below adding `spans = spans[:n]`